Assets/HoloToolkit/Input/Scripts/Gaze/GazeManager.cs

```diff
@@ -294,25 +295,19 @@ private void RaycastUnityUI()
         private RaycastResult FindClosestRaycastHitInLayerMasks(List<RaycastResult> candidates, LayerMask[] layerMaskList)
         {
             int combinedLayerMask = 0;
-            for (int i = 0; i < layerMaskList.Length; i++)
+            for (var i = 0; i < layerMaskList.Length; i++)
             {
                 combinedLayerMask = combinedLayerMask | layerMaskList[i].value;
             }
 
-            RaycastResult? minHit = null;
-            for (var i = 0; i < candidates.Count; ++i)
-            {
-                if (candidates[i].gameObject == null || !IsLayerInLayerMask(candidates[i].gameObject.layer, combinedLayerMask))
-                {
-                    continue;
-                }
-                if (minHit == null || candidates[i].distance < minHit.Value.distance)
-                {
-                    minHit = candidates[i];
-                }
-            }
+            var canvasCandidates = candidates.Where(c => c.gameObject != null)
```

Comment:
Is there a way to do this without all the allocations from linq?

Reply:
What do you mean with by allocations from linq

Reply:
We've got a closure allocation on line 297, and multiple object and delegate allocations occurring in the linq statements. Generally good to avoid linq in update loops, if possible.